src/harness/reference_models/pre_iap_filtering/fss_purge.py

```diff
@@ -0,0 +1,296 @@
+  else:
+      cbsd_height_agl = cbsd_height
+
+  # Invoking ITM propagation model
+  db_loss, incidence_angle, _ = wf_itm.CalcItmPropagationLoss(
+                                         cbsd_latitude,
+                                         cbsd_longitude,
+                                         cbsd_height_agl,
+                                         fss.latitude,
+                                         fss.longitude,
+                                         fss.height_agl,
+                                         cbsd_indoor_dep,
+                                         reliability=-1,
+                                         freq_mhz=interference.FREQ_PROP_MODEL_MHZ)
+
+  return db_loss, incidence_angle
+
+def convertAmslToAgl(latitude, longitude, height_amsl):
+  """ Convert AMSL height to AGL height."""
+
+  altitude_cbsd = wf_itm.terrainDriver.GetTerrainElevation(latitude, longitude)
+  return height_amsl - altitude_cbsd
+
+def getAntennaGainFssReceiver(fss_entity, incidence_angle):
+  """Get antenna gain of FSS receiver determined using FSS antenna reference model.
+  Args:
+    cbsd: A CbsdData object
+    incidence_angle: Incidence angle as identified by ITM model.
+  Returns:
+    Antenna gain of FSS in dBi.
+  """
+
+  #Invoke FSS antenna reference model.
+  fss_ant_gain = antenna.GetFssAntennaGains(
+                              incidence_angle.hor_rx, incidence_angle.ver_rx,
+                              fss_entity.pointing_azimuth,
+                              fss_entity.pointing_elevation,
+                              fss_entity.max_gain_dbi)
+  return fss_ant_gain
+
+def fssPurgeReferenceModel(sas_uut_fad, sas_test_harness_fads, fss_records):
+  """Entry point function to execute FSS purge list model.
+
+  Performs FSS purge model as described in R2-SGN-29 on the FAD CBSD records of SAS UUT
+  and SAS test harnesses for a list of FSS records. The grants are removed from the CBSD
+  records of the FAD objects.
+  Args:
+    sas_uut_fad: A FullActivityDump object containing the FAD records of SAS UUT.
+    sas_test_harness_fads: A list of FullActivityDump objects containing the FAD records 
+      from SAS test harnesses.
+    fss_records: A list of FSS record dictionary.
+  """
+  # Get the CBSD list from the FAD Object. 
+  cbsds = []
+  cbsds.extend(sas_uut_fad.getCbsdRecords())
+  for fad in sas_test_harness_fads:
+    cbsds.extend(fad.getCbsdRecords())
+
+  grants_to_purged_for_all_fss = []
+  for fss_record in fss_records:
+    # If the FSS is of TT&C type then perform the FSS purge model for the FSS.
+    if fss_record['ttc']:
+      neighboring_cbsds_with_grants = pre_iap_util.getFssNeighboringCbsdsWithGrants(cbsds, fss_record, 40)
+      if neighboring_cbsds_with_grants:
+        fss_entity = getFssInfo(fss_record)
+        grants_to_purge_for_fss = performPurge(neighboring_cbsds_with_grants, fss_entity)
+        grants_to_purged_for_all_fss.extend(grants_to_purge_for_fss)
```

Comment:
Duplicate grant entries (due to a grant being common to multiple FSS) are not removed from the grants_to_purged_for_fss list. This could cause the list to become huge. Shouldn't the duplicates be pruned before using the list for futher processing?

Reply:
As   suggested, modified the code to not create duplicate entries